### PR TITLE
Update documentation on plugin installation and management

### PR DESCRIPTION
**Pull request by fmoralesc**
When working on #3125 I noticed the plugin installation and management in usr_05.txt was a bit outdated and kinda hacked up, so I tried to improve it.

### CODE REVIEW

File: runtime/doc/usr_05.txt [hunk 1]
```diff
@@ -250,61 +250,87 @@ The ":map" command (with no arguments) lists your current mappings.  At
...
-    global plugin: Used for all kinds of files
-  filetype plugin: Only used for a specific type of file
-
-The global plugins will be discussed first, then the filetype ones
-|add-filetype-plugin|.
-
-
-GLOBAL PLUGINS						*standard-plugin*
-
+*05.4*	Plugins					*add-plugin* *plugin*
```
Comment (ZyX-I) [inline runtime/doc/usr_05.txt]: When you rename a section of the user manual you need to rename it also at the top of the file and in `usr_toc.txt`.

Comment (fmoralesc) [inline runtime/doc/usr_05.txt]: :+1:

File: runtime/doc/usr_05.txt [hunk 2]
```diff
@@ -250,61 +250,87 @@ The ":map" command (with no arguments) lists your current mappings.  At
...
   and its sub-directories.
-- Download from the net.  There is a large collection on http://www.vim.org.
+- In the net:
+  - A nicely curated list of plugins is available at http://vimawesome.com/.
+  - There is a large collection of plugins on http://www.vim.org/scripts.
+  - Github is very popular for plugin creators. See
+    https://github.com/search?q=language%3AVimL&type=Repositories
 - They are sometimes posted in a Vim |maillist|.
-- You could write one yourself, see |write-plugin|.
+- You could write one yourself ;) (see |write-plugin|).
```
Comment (ZyX-I) [inline runtime/doc/usr_05.txt]: Does not smiley end a sentense? I would write the ending as `See |write-plugin|.`

File: runtime/doc/usr_05.txt [hunk 3]
```diff
@@ -250,61 +250,87 @@ The ":map" command (with no arguments) lists your current mappings.  At
...
+
+
```
Comment (ZyX-I) [inline runtime/doc/usr_05.txt]: Previous section was prefixed by a single newline.

File: runtime/doc/usr_05.txt [hunk 4]
```diff
@@ -250,61 +250,87 @@ The ":map" command (with no arguments) lists your current mappings.  At
...
-	Mac OS X	~/.vim/plugin/
+For simple plugins, it should be enough to drop their files in the appropriate
+directories.
 
-Example for Unix (assuming you didn't have a plugin directory yet): >
+For example, to install the justify.vim plugin in *nix (assuming you didn't
+have a plugin directory yet): >
 
 	mkdir ~/.vim
 	mkdir ~/.vim/plugin
```
Comment (ZyX-I) [inline runtime/doc/usr_05.txt]: This is ~/.nvim.

File: runtime/doc/usr_05.txt [hunk 5]
```diff
@@ -313,38 +339,12 @@ Example for Unix (assuming you didn't have a plugin directory yet): >
 That's all!  Now you can use the commands defined in this plugin to justify
 text.
 
-Instead of putting plugins directly into the plugin/ directory, you may
-better organize them by putting them into subdirectories under plugin/.
-As an example, consider using "~/.vim/plugin/perl/*.vim" for all your Perl
+Instead of putting plugins directly into the plugin/ directory, you may better
+organize them by putting them into subdirectories under plugin/.  As an
+example, consider using "~/.vim/plugin/git/*.vim" for all the Git-related
```
Comment (ZyX-I) [inline runtime/doc/usr_05.txt]: ~/.nvim.

File: runtime/doc/usr_05.txt [hunk 6]
```diff
@@ -387,8 +383,91 @@ Only files of this filetype will use the settings from the plugin.  The <name>
 part of the plugin file doesn't matter, you can use it to have several plugins
 for the same filetype.  Note that it must end in ".vim".
 
+USING PLUGIN BUNDLES                                      *plugin-bundles*
+
+The manual method described above has the problem that things can get messy
+once you install a few plugins, specially if they include more than one file
+(which is common). A way to solve this is to install plugins in their own
+directories and then add the path of those to the 'runtimepath' setting (so
+Vim will use the files from those directories when looking for plugins). In
```
Comment (ZyX-I) [inline runtime/doc/usr_05.txt]: Two spaces after a dot.

File: runtime/doc/usr_05.txt [hunk 7]
```diff
@@ -387,8 +383,91 @@ Only files of this filetype will use the settings from the plugin.  The <name>
...
+practice, this means that instead of having a |vimfiles| tree like >
+
+	~.vim/
```
Comment (ZyX-I) [inline runtime/doc/usr_05.txt]: `~/.nvim`, not `~.vim`.

File: runtime/doc/usr_05.txt [hunk 8]
```diff
@@ -387,8 +383,91 @@ Only files of this filetype will use the settings from the plugin.  The <name>
...
+		a.vim
+	    ftplugin/
+		b.vim
+		e.vim
+	    syntax/
+		b.vim
+
+you can have something like this >
+
+	~.vim/
```
Comment (ZyX-I) [inline runtime/doc/usr_05.txt]: `~/.nvim`

File: runtime/doc/usr_05.txt [hunk 9]
```diff
@@ -387,8 +383,91 @@ Only files of this filetype will use the settings from the plugin.  The <name>
...
+		    ftplugin/
+			b.vim
+		    syntax/
+			b.vim
+		e/
+		    ftplugin/
+			e.vim
+
+This method allows the plugins to be distributed as their own trees, so they
+can be installed without affecting other plugins. This is very helpful when
```
Comment (ZyX-I) [inline runtime/doc/usr_05.txt]: Two spaces after dot.

File: runtime/doc/usr_05.txt [hunk 10]
```diff
@@ -387,8 +383,91 @@ Only files of this filetype will use the settings from the plugin.  The <name>
...
+	plugin/
+	    awesome.vim
+	doc/
+	    awesome.txt
+
+from http://github.com/developer/awesome-vim, they can do so by (assuming there
+is no "bundles" directory)
+
+    1. creating the plugin directory >
+	$ mkdir -p ~/.vim/bundles/awesome-vim
```
Comment (ZyX-I) [inline runtime/doc/usr_05.txt]: ~/.nvim

File: runtime/doc/usr_05.txt [hunk 11]
```diff
@@ -387,8 +383,91 @@ Only files of this filetype will use the settings from the plugin.  The <name>
...
+<    2. copying the files into the "bundles/awesome-vim/" folder >
+	$ cp awesome-vim/* ~/.vim/bundles/awesome-vim/
```
Comment (ZyX-I) [inline runtime/doc/usr_05.txt]: ~/.nvim

File: runtime/doc/usr_05.txt [hunk 12]
```diff
@@ -387,8 +383,91 @@ Only files of this filetype will use the settings from the plugin.  The <name>
...
+<    3. adding >
+
+	set runtimepath+=~/.vim/bundles/awesome-vim
```
Comment (ZyX-I) [inline runtime/doc/usr_05.txt]: ~/.nvim. Also unlike other code blocks this one contains empty lines before and after which is strange.

File: runtime/doc/usr_05.txt [hunk 13]
```diff
@@ -387,8 +383,91 @@ Only files of this filetype will use the settings from the plugin.  The <name>
...
+
+<       to their |vimrc|.
+
+Most commonly, instead of steps 1-2, this should work: >
```
Comment (ZyX-I) [inline runtime/doc/usr_05.txt]: Not sure, but should not it worded as “Most common”?

File: runtime/doc/usr_05.txt [hunk 14]
```diff
@@ -387,8 +383,91 @@ Only files of this filetype will use the settings from the plugin.  The <name>
...
+The most popular plugin managers are:
+
+- vim-plug https://github.com/junegunn/vim-plug
+- Vundle https://github.com/VundleVim/Vundle.vim
+- neobundle.vim https://github.com/Shougo/neobundle.vim
+- VAM https://github.com/MarcWeber/vim-addon-manager
+
+Please refer to their documentation on how to use them.
+
+
```
Comment (ZyX-I) [inline runtime/doc/usr_05.txt]: One empty line.

File: runtime/doc/usr_05.txt [hunk 15]
```diff
@@ -387,8 +383,91 @@ Only files of this filetype will use the settings from the plugin.  The <name>
...
+	$ cp awesome-vim/* ~/.vim/bundles/awesome-vim/
+<    3. adding >
+
+	set runtimepath+=~/.vim/bundles/awesome-vim
+
+<       to their |vimrc|.
+
+Most commonly, instead of steps 1-2, this should work: >
+
+    $ mkdir ~/.vim/bundles/
```
Comment (ZyX-I) [inline runtime/doc/usr_05.txt]: ~/.nvim

File: runtime/doc/usr_05.txt [hunk 16]
```diff
@@ -387,8 +383,91 @@ Only files of this filetype will use the settings from the plugin.  The <name>
...
+    $ cd ~/.vim/bundles
```
Comment (ZyX-I) [inline runtime/doc/usr_05.txt]: ~/.nvim

File: runtime/doc/usr_05.txt [hunk 17]
```diff
@@ -387,8 +383,91 @@ Only files of this filetype will use the settings from the plugin.  The <name>
...
+    $ git clone https://github.com/developer/awesome-vim
+
+
+PLUGIN MANAGERS                                            *plugin-managers*
+
+Bundle-style plugins can be handled by so-called 'plugin managers', which are
```
Comment (ZyX-I) [inline runtime/doc/usr_05.txt]: `"plugin manager"`. `''` is for options

File: runtime/doc/usr_05.txt (hunk 12 above)
Comment (fmoralesc) [inline runtime/doc/usr_05.txt]: Yeah, I thought it looked weird too close to the surrounding lines since this is not prefixed by a `$`, like in the other cases. It looks clearer this way to me. I could give more room to the previous codeblocks, but I won't remove the empty lines in this one.

File: runtime/doc/usr_05.txt [hunk 18]
```diff
@@ -250,61 +250,87 @@ The ":map" command (with no arguments) lists your current mappings.  At
...
-							*add-global-plugin*
-You can add a global plugin to add functionality that will always be present
-when you use Vim.  There are only two steps for adding a global plugin:
-1. Get a copy of the plugin.
-2. Drop it in the right directory.
+Global plugins reside under the `plugin/` |vimfiles| directory.
+
+                                                        *ftplugins*
+Filetype plugins ("ftplugins") are only used for a specific type of file, and
+usually change the buffer-local settings. For example, the standard vim C
```
Comment (ZyX-I) [inline runtime/doc/usr_05.txt]: Two spaces after a dot.

File: runtime/doc/usr_05.txt [hunk 19]
```diff
@@ -250,61 +250,87 @@ The ":map" command (with no arguments) lists your current mappings.  At
...
+ftplugin sets 'formatoptions' to break comment lines but not other lines, and
+to insert the comment leader when hitting <CR> or using |o|. Filetype plugins
```
Comment (ZyX-I) [inline runtime/doc/usr_05.txt]: Two spaces.

File: runtime/doc/usr_05.txt [hunk 20]
```diff
@@ -250,61 +250,87 @@ The ":map" command (with no arguments) lists your current mappings.  At
...
+You can add a plugin very easily by dropping it in your |vimfiles| directory,
+but there are more advanced ways to manage plugins, which will be described
+below.
+
+KINDS OF PLUGIN                                         *plugin-types*
+
+There are mainly two kinds of plugin: global and filetype-specific.
+
+Global plugins are sourced on startup and usually add functionality that will
+be available regardless of the opened files. For example, the |matchparen|
```
Comment (ZyX-I) [inline runtime/doc/usr_05.txt]: Two spaces.

File: runtime/doc/usr_05.txt (hunk 12 above)
Comment (ZyX-I) [inline runtime/doc/usr_05.txt]: 29.09.2015, 01:36, "Felipe Morales" <notifications@github.com>:
> In runtime/doc/usr_05.txt:
>
>> + awesome.vim > + plugin/ > + awesome.vim > + doc/ > + awesome.txt > + > +from http://github.com/developer/awesome-vim, they can do so by (assuming there > +is no "bundles" directory) > + > + 1. creating the plugin directory > > + $ mkdir -p ~/.vim/bundles/awesome-vim > +< 2. copying the files into the "bundles/awesome-vim/" folder > > + $ cp awesome-vim/* ~/.vim/bundles/awesome-vim/ > +< 3. adding > > + > + set runtimepath+=~/.vim/bundles/awesome-vim
>
> Yeah, I thought it looked weird too close to the surrounding lines since this is not prefixed by a $, like in the other cases. It looks clearer this way to me. I could give more room to the previous codeblocks, but I won't remove the empty lines in this one.

It does not matter as long as you are consistent. If you think that less space here looks bad, add more space there.

File: runtime/doc/usr_05.txt (hunk 13 above)
Comment (fmoralesc) [inline runtime/doc/usr_05.txt]: "commonly" is working as an adverb here, I think "most common" wouldn't be right.

File: runtime/doc/usr_05.txt [hunk 21]
```diff
@@ -378,17 +373,102 @@ The generic names for the filetype plugins are: >
...
+
+	$ cp awesome-vim/* ~/.nvim/bundles/awesome-vim/
+
+<    3. adding >
+
+	set runtimepath+=~/.nvim/bundles/awesome-vim
+
+<       to their |vimrc|.
+
+Most commonly, instead of steps 1-2, this should work: >
```
Comment (ZyX-I) [inline runtime/doc/usr_05.txt]: Adverb which modifies what? In this position with these commas this looks like an introductory phrase (where adverb has nothing to modify because this is rather independent) and not something that modifies verb “work”. I would rephrase this as “Most likely steps 1 and 2 can be replaced with” because I am pretty sure “instead of steps 1-2” could not be placed in that position.

Comment (fmoralesc) [inline runtime/doc/usr_05.txt]: Sure.

**Comment by ZyX-I** [comment]
LGTM

**Comment by justinmk** [comment]
I liked the brevity of the old copy. Also not sure about some of the additional tags. Let's set this for 0.2.

File: runtime/doc/usr_05.txt [hunk 22]
```diff
@@ -250,121 +250,115 @@ The ":map" command (with no arguments) lists your current mappings.  At
...
-
-
-GLOBAL PLUGINS						*standard-plugin*
-
+*05.4*	Plugins					*add-plugin* *plugin*
+
+Vim's functionality can be extended by adding plugins.  A plugin is
+essentially a set of Vim script files that are loaded automatically when Vim
+starts or on some other event (like opening a file of a certain type).
+Plugins can be used to add support to more programming languages, extend Vim's
```
Comment (brcolow) [inline runtime/doc/usr_05.txt]: Should read: 
Plugins can be used to add support *for* more programming languages,

File: runtime/doc/usr_05.txt [hunk 23]
```diff
@@ -250,121 +250,115 @@ The ":map" command (with no arguments) lists your current mappings.  At
...
+command "vocabulary", simplify some use-cases, add some bling to Vim, and even
+turn it into a different sort of application.  Thankfully, the community has
+made available on the net a great ammount of plugins that any user can benefit
```
Comment (brcolow) [inline runtime/doc/usr_05.txt]: Maybe this should read:

"Thankfully there is a vibrant and active community of plugin authors from around the globe that every user can benefit from."

File: runtime/doc/usr_05.txt [hunk 24]
```diff
@@ -250,121 +250,115 @@ The ":map" command (with no arguments) lists your current mappings.  At
...
+from.
+
+You can add a plugin very easily by dropping it in your |vimfiles| directory,
+but there are more advanced ways to manage plugins, which will be described
```
Comment (brcolow) [inline runtime/doc/usr_05.txt]: Instead of advanced, maybe convenient would be a better fit.

File: runtime/doc/usr_05.txt [hunk 25]
```diff
@@ -250,121 +250,115 @@ The ":map" command (with no arguments) lists your current mappings.  At
...
+below.
+
+KINDS OF PLUGIN                                         *plugin-types*
+
+There are mainly two kinds of plugin: global and filetype-specific.
```
Comment (brcolow) [inline runtime/doc/usr_05.txt]: Should read: There are mainly two kinds of plugin*s*:....

File: runtime/doc/usr_05.txt [hunk 26]
```diff
@@ -250,121 +250,115 @@ The ":map" command (with no arguments) lists your current mappings.  At
...
+
+Global plugins are sourced on startup and usually add functionality that will
+be available regardless of the opened files.  For example, the |matchparen|
```
Comment (brcolow) [inline runtime/doc/usr_05.txt]: Maybe should read:

Global plugins are sourced on startup and usually add functionality that is available regardless of which files are open.

File: runtime/doc/usr_05.txt [hunk 27]
```diff
@@ -250,121 +250,115 @@ The ":map" command (with no arguments) lists your current mappings.  At
...
-1. Get a copy of the plugin.
-2. Drop it in the right directory.
+Global plugins reside under the `plugin/` |vimfiles| directory.
+
+                                                        *ftplugins*
+Filetype plugins ("ftplugins") are only used for a specific type of file, and
+usually change the buffer-local settings.  For example, the standard vim C
+ftplugin sets 'formatoptions' to break comment lines but not other lines, and
+to insert the comment leader when hitting <CR> or using |o|.  Filetype plugins
+are usually distributed alongside syntax files, and indent and filetype
```
Comment (brcolow) [inline runtime/doc/usr_05.txt]: I don't like the double and here, as in "...usually distributed alongside syntax files, *and* indent *and* filetype detection scripts..." but I am not sure how to fix it.

File: runtime/doc/usr_05.txt [hunk 28]
```diff
@@ -250,121 +250,115 @@ The ":map" command (with no arguments) lists your current mappings.  At
...
+detection scripts to provide support for aditional programming or markup
+languages in Vim.
+
+Filetype plugins reside under the `ftplugin/` |vimfiles| directory.
+
+The Vim distribution comes with a set of plugins for different filetypes that
+you can start using with this command: >
```
Comment (brcolow) [inline runtime/doc/usr_05.txt]: Not sure about this one, but maybe it should read:

you can start using with the '>' command

(not sure about single quotes format)

File: runtime/doc/usr_05.txt [hunk 29]
```diff
@@ -250,121 +250,115 @@ The ":map" command (with no arguments) lists your current mappings.  At
...
+GETTING PLUGINS                                         *plugins-getting*
 
-GETTING A GLOBAL PLUGIN
+There are many places to get plugins from:
 
-Where can you find plugins?
 - Some come with Vim.  You can find them in the directory $VIMRUNTIME/macros
   and its sub-directories.
-- Download from the net.  There is a large collection on http://www.vim.org.
+- In the net:
```
Comment (brcolow) [inline runtime/doc/usr_05.txt]: Should read:

On the net:

File: runtime/doc/usr_05.txt [hunk 30]
```diff
@@ -250,121 +250,115 @@ The ":map" command (with no arguments) lists your current mappings.  At
...
-- You could write one yourself, see |write-plugin|.
+- You could write one yourself ;) See |write-plugin|.
+
+INSTALLING PLUGINS                   *add-global-plugin*  *add-filetype-plugin*
 
 Some plugins come as a vimball archive, see |vimball|.
 Some plugins can be updated automatically, see |getscript|.
 
+For all other cases, you should be able to follow the guidelines below (also,
+it's very likely that the plugin creators has given instructions on how to
```
Comment (brcolow) [inline runtime/doc/usr_05.txt]: Should read:

it's very likely that the plugins creators *have* given instructions on how...

File: runtime/doc/usr_05.txt [hunk 31]
```diff
@@ -250,121 +250,115 @@ The ":map" command (with no arguments) lists your current mappings.  At
...
-First read the text in the plugin itself to check for any special conditions.
-Then copy the file to your plugin directory:
+For simple plugins, it should be enough to drop their files in the appropriate
+directories.
 
-	system		plugin directory ~
-	Unix		~/.vim/plugin/
-	Macintosh	$VIM:vimfiles:plugin
-	Mac OS X	~/.vim/plugin/
+For example, to install the justify.vim plugin in *nix (assuming you didn't
```
Comment (brcolow) [inline runtime/doc/usr_05.txt]: in *nix -> on *nix

File: runtime/doc/usr_05.txt [hunk 32]
```diff
@@ -378,17 +372,102 @@ The generic names for the filetype plugins are: >
...
+It is very likely that steps 1-2 can be replaced with something like: >
+
+    $ mkdir -p ~/.nvim/bundles/
+    $ cd ~/.nvim/bundles
+    $ git clone https://github.com/developer/awesome-vim
+
+PLUGIN MANAGERS                                            *plugin-managers*
+
+Bundle-style plugins can be handled by so-called "plugin managers", which are
+special plugins whose purpose is to handle 'runtimepath' and help the user to
```
Comment (brcolow) [inline runtime/doc/usr_05.txt]: Instead of "special plugins whose purpose is to..." maybe "special purpose plugins that handle 'runtimepath'..."

File: runtime/doc/usr_05.txt [hunk 33]
```diff
@@ -378,17 +372,102 @@ The generic names for the filetype plugins are: >
...
+install, update and load plugins.
+
+The most popular plugin managers are:
```
Comment (brcolow) [inline runtime/doc/usr_05.txt]: Instead of "The most popular", for reasons of maintenance and future-proofing, maybe we should say "Some great plugin managers are" or "Recommended plugin managers are" or something like that.

**Comment by fmoralesc** [comment]
Thanks, @brcolow, I'll include most (if not all) of your suggestions.

**Comment by jrobeson** [comment]
doesn't this require changes due to the XDG PR being merged?

**Comment by fmoralesc** [comment]
@jrobeson Sure does, thanks for the reminder.

**Comment by justinmk** [comment]
Might need to revisit this in light of and friends.

**Comment by justinmk** [comment]
I tried to merge this, but there's too much that's out of date. And the section on "bundles" is now replaced by Vim 8 "packages".
